src/AppBundle/Controller/MembershipController.php

```diff
@@ -33,18 +36,67 @@ public function registerAction(Request $request): Response
 
             $this->addFlash('info', $this->get('translator')->trans('adherent.registration.success'));
 
-            return $this->redirectToRoute('app_membership_register');
+            $donation = $this->get('app.donation.factory')->createDonationFromAdherent($adherent);
+
+            $session = $request->getSession();
+            $session->set('registering_donation', $donation);
```

Comment:
I would move this logic in a DonationUtils (or something similar) service like the `AuthenticationUtils` service in Symfony.